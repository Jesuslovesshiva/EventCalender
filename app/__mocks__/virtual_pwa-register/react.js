import {useState} from 'react'

const useRegisterSW = jest.fn(() => {
  const [needRefresh, setNeedRefresh] = useState(true)

  return {
    needRefresh: [needRefresh, setNeedRefresh],
    updateServiceWorker: () => {}
  }
})

// eslint-disable-next-line import/prefer-default-export
export {useRegisterSW}
