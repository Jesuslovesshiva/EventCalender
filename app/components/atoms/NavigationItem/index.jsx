import {Link} from '@tanstack/react-location'
import classNames from 'classnames'
import {noop} from 'lodash'
import PropTypes from 'prop-types'
import React from 'react'

import ArrowDropDownIcon from '@/components/atoms/Icon/ArrowDropDown'

/* eslint-disable sort-keys */
const LEVELS = {
  ONE: 'one',
  TWO: 'two',
  THREE: 'three'
}
/* eslint-enable sort-keys */

const getLevelClasses = ({isDisabled, isOpen, level}) => {
  const CLASSES = {
    [LEVELS.ONE]: [
      'text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t',
      'focus-visible:ring-white',
      'hover:text-primary hover:bg-white hover:shadow',
      {'text-primary bg-white shadow': isOpen}
    ],
    [LEVELS.TWO]: [
      'text-[11px] leading-[13px] text-gray-dark px-2.5 pt-[5px] pb-[7px] rounded-t',
      {
        'text-gray-dark': isDisabled,
        'text-primary': !isDisabled
      },
      'focus-visible:ring-primary',
      'hover:bg-white hover:shadow',
      {'bg-white shadow': isOpen}
    ],
    [LEVELS.THREE]: [
      'text-[11px] leading-[13px] text-primary px-2.5 pt-1 pb-[5px] rounded',
      'focus-visible:ring-primary',
      'hover:bg-silver'
    ]
  }

  return CLASSES[level]
}

const NavigationItem = React.forwardRef((props, ref) => (
  <Link
    className={classNames(
      'font-bold inline-flex flex-col transition motion-reduce:transition-none',
      'focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset',
      {'cursor-default': props.isDisabled},
      getLevelClasses(props)
    )}
    tabIndex="0"
    to={props.to}
    disabled={props.isDisabled}
    onClick={props.onClick}
    onKeyDown={props.onKeyDown}
    onMouseEnter={props.onMouseEnter}
    onMouseLeave={props.onMouseLeave}
    _ref={ref}
  >
    {({isActive}) => (
      <>
        <div className="flex items-center">
          {props.children}
          {props.hasSublevel && <ArrowDropDownIcon className="text-lg" />}
        </div>

        <div className="relative">
          <div
            className={classNames(
              'w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current',
              'motion-reduce:transition-none transition-[opacity,transform]',
              {
                'mt-px': props.level === LEVELS.ONE,
                'opacity-0 translate-y-0.5': !isActive,
                'opacity-100 translate-y-0': isActive
              }
            )}
          />
        </div>
      </>
    )}
  </Link>
))

NavigationItem.displayName = 'NavigationItem'

NavigationItem.propTypes = {
  children: PropTypes.node,
  hasSublevel: PropTypes.bool,
  isDisabled: PropTypes.bool,
  isOpen: PropTypes.bool,
  level: PropTypes.oneOf(Object.values(LEVELS)),
  onClick: PropTypes.func,
  onKeyDown: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
  to: PropTypes.string
}

NavigationItem.defaultProps = {
  children: null,
  hasSublevel: false,
  isDisabled: false,
  isOpen: false,
  level: LEVELS.ONE,
  onClick: noop,
  onKeyDown: noop,
  onMouseEnter: noop,
  onMouseLeave: noop,
  to: null
}

export {LEVELS}
export default NavigationItem
