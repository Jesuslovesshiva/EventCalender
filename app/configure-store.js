import {
  combineReducers,
  configureStore as configureToolkitStore,
  isPlain
} from '@reduxjs/toolkit'
import {createBrowserHistory} from '@tanstack/react-location'
import {createReduxHistoryContext} from 'redux-first-history'
import {createInjectorsEnhancer} from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'

import ActionSubscription, {
  actionListenerMiddleware
} from '@/components/organisms/ActionSubscription'

const actionEmitter = new ActionSubscription.ActionEmitter()

const {createReduxHistory, routerMiddleware, routerReducer} =
  createReduxHistoryContext({
    history: createBrowserHistory()
  })

const sagaMiddleware = createSagaMiddleware()

const configureStore = () => {
  const {run: runSaga} = sagaMiddleware
  const middlewares = [
    sagaMiddleware,
    routerMiddleware,
    actionListenerMiddleware(actionEmitter)
  ]

  const createReducer = injectedReducers =>
    combineReducers({
      router: routerReducer,
      ...injectedReducers
    })
  const enhancers = [createInjectorsEnhancer({createReducer, runSaga})]

  const isSerializable = value =>
    value instanceof Error ||
    value instanceof File ||
    value instanceof Response ||
    isPlain(value)

  const store = configureToolkitStore({
    // eslint-disable-next-line no-undef
    devTools: process.env.NODE_ENV !== 'production',
    enhancers,
    middleware: getDefaultMiddleware => [
      ...getDefaultMiddleware({serializableCheck: {isSerializable}}),
      ...middlewares
    ],
    reducer: createReducer()
  })

  store.injectedReducers = {}
  store.injectedSagas = {}

  return store
}

const store = configureStore()
const history = createReduxHistory(store)

export {configureStore, history, actionEmitter}

export default store
