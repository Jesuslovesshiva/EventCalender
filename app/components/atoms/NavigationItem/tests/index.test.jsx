import {fireEvent, waitFor} from '@testing-library/react'

import {history} from '@/configure-store'
import render from '@/utils/test-utils/custom-render'

import NavigationItem from '../index'

jest.mock('@/components/atoms/Icon/ArrowDropDown', () => () => (
  <div data-testid="ArrowDropDownIcon" />
))

const defaultProps = {
  children: 'Navigation Item'
}

const options = {wrapperProps: {hasRouter: true}}
const renderComponent = props =>
  render(<NavigationItem {...defaultProps} {...props} />, options)

describe('NavigationItem', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a caret', () => {
    const {getByTestId} = renderComponent({hasSublevel: true})
    expect(getByTestId('ArrowDropDownIcon')).toBeInTheDocument()
  })

  it('should navigate on click', async () => {
    const {container} = renderComponent({to: '/baz'})
    fireEvent.click(container.firstChild)
    await waitFor(() => expect(history.location.pathname).toBe('/baz'))
  })

  it('should not navigate on click if disabled', () => {
    const onClick = jest.fn()
    const {container} = renderComponent({
      isDisabled: true,
      onClick,
      to: '/foo/bar'
    })
    fireEvent.click(container.firstChild)
    expect(onClick).not.toHaveBeenCalled()
    expect(container.firstChild.href).toBe('')
  })
})
