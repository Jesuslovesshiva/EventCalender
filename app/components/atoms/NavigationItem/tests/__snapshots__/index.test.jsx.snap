// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem should match the snapshot 1`] = `
<a
  class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow active"
  href="/"
  tabindex="0"
>
  <div
    class="flex items-center"
  >
    Navigation Item
  </div>
  <div
    class="relative"
  >
    <div
      class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-100 translate-y-0"
    />
  </div>
</a>
`;
